crates/lockfile/src/package_snapshot.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{LockfileResolution, PkgVerPeer};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct LockfilePeerDependencyMetaValue {
+    optional: bool,
+}
+
+// Reference: https://github.com/pnpm/pnpm/blob/main/lockfile/lockfile-file/src/sortLockfileKeys.ts#L5
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PackageSnapshot {
+    pub resolution: LockfileResolution,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub id: Option<String>,
+
+    pub name: Option<String>,
+    pub version: Option<String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub engines: Option<HashMap<String, String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub cpu: Option<Vec<String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub os: Option<Vec<String>>,
+    // TODO: Add `libc`
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub deprecated: Option<bool>,
```

Comment:
In this lockfile this won't be a boolean. It will be the text of the deprecation message.